src/pepr/operator/controllers/utils.ts

```diff
@@ -38,3 +40,20 @@ export function getOwnerRef(cr: GenericKind): V1OwnerReference[] {
     },
   ];
 }
+
+export async function purgeOrphans<T extends GenericClass>(
+  generation: string,
+  namespace: string,
+  pkgName: string,
+  kind: T,
+  log: Logger,
+) {
+  const resources = await K8s(kind).InNamespace(namespace).WithLabel("uds/package", pkgName).Get();
+
+  for (const resource of resources.items) {
+    if (resource.metadata?.labels?.["uds/generation"] !== generation) {
+      log.debug(resource, `Deleting orphaned ${resource.kind!} ${resource.metadata!.name}`);
+      await K8s(kind).Delete(resource);
+    }
+  }
```

Comment:
This seems fine, just curious if there was a particular intent behind the switch from filter -> loop over only items to delete vs loop over all -> conditional delete?

Reply:
It was not for any particular reason, I copied the code from the most recent example in front of me which was the authservice code. 

I think technically this is ever so slightly better, avoiding creating another array, but I could go either way if you think the other way is more idiomatic/readable.